go/storage/mkvs/urkel/fuzz/tree.go

```diff
@@ -166,16 +229,25 @@ func (t *TreeFuzz) fail(format string, a ...interface{}) {
 	// In case there is a failure, dump the operation history so it can be used to create a test
 	// vector for unit tests.
 	fmt.Printf("--- FAILURE: Dumping operation history ---\n")
+
 	history, _ := json.MarshalIndent(t.history, "", "    ")
-	fmt.Printf("%s\n", history)
+	f, err := ioutil.TempFile("", "oasis-node-fuzz-mkvs-dump-*.json")
+	if err == nil {
+		_, _ = f.Write(history)
+		f.Close()
+
+		fmt.Printf("[see %s]\n", f.Name())
+	} else {
+		fmt.Printf("[unable to save dump: %s]", err.Error())
+	}
 	fmt.Printf("------------------------------------------\n")
 
 	panic(fmt.Sprintf(format, a...))
 }
 
 func NewTreeFuzz() (*TreeFuzz, *commonFuzz.InterfaceFuzzer) {
```

Comment:
Yes this is special because it is stateful, this is also why we need to explicitly record history.

The problem is that otherwise go-fuzz would need to generate long histories of operations -- when I tried this variant it didn't go so well with discovering issues and the current stateful variant performs much better.

Reply:
> when I tried this variant it didn't go so well with discovering issues

alright, guess I'll resolve this thread